lib/ably/realtime/channel/channel_manager.rb

```diff
@@ -226,11 +232,6 @@ def send_state_change_protocol_message(new_state, state_if_failed, message_optio
           end
         end
 
-        channel.once_state_changed do
-          @pending_state_change_timer.cancel if @pending_state_change_timer
-          @pending_state_change_timer = nil
-        end
-
         resend_if_disconnected_and_connected = lambda do
           connection.unsafe_once(:disconnected) do
             next unless pending_state_change_timer
```

Comment:
@lawrence-forooghian I didn't touch the existing code as such.
But I am having hard time understanding usage of `next unless` here.
Since, next is only used inside a loop, I think it is being used to inform internal emitter callback execution loop to execute next callback when condition holds true. Basically, next lines of code is skipped when `pending_state_change_timer` is nil. I think it can be simplified using simple `if` condition.

Reply:
I have kept changes small for better understanding. Maybe, I can refactor the code as a part of next commit.

Reply:
https://github.com/ably/ably-ruby/pull/416/commits/1324a72a858910813514ae65bd66dd29b13adff7